### PR TITLE
RUN-1050: Add reference counting information to assertion in DropPeer

### PR DESCRIPTION
**Pull request by sgeisenh**
https://linear.app/replay/issue/RUN-1050#comment-e25ee2d1

### CODE REVIEW

File: mojo/core/node_controller.cc [hunk 1]
```diff
@@ -655,8 +656,20 @@ void NodeController::DropPeer(const ports::NodeName& node_name,
     auto it = peers_.find(name);
 
     if (it != peers_.end()) {
+      auto channel_has_one_ref = it->second.get()->HasOneRef();
+
+      // Since some amount of work happens after this point, there's no
+      // guarantee that this assert will fire even if there is a thread
+      // interleaving issue present.
+      //
+      // Based on the comment above, it seems likely that we might be able to
```
Comment (Domiii) [inline mojo/core/node_controller.cc:665]: Nitpick: Are you referring to the comment you mention in the Linear comment [here](https://linear.app/replay/issue/RUN-1050#comment-e25ee2d1)? Do you want to reference it directly? (Else, this is hard to understand)

File: mojo/core/node_controller.cc [hunk 2]
```diff
@@ -655,8 +656,20 @@ void NodeController::DropPeer(const ports::NodeName& node_name,
     auto it = peers_.find(name);
 
     if (it != peers_.end()) {
+      auto channel_has_one_ref = it->second.get()->HasOneRef();
```
Comment (Domiii) [inline mojo/core/node_controller.cc:659]: Oh this is nice!

**Review by Domiii — APPROVED** [review]
It appears, our CI is having a bit of a hicc-up (I brought it up [here](https://discord.com/channels/779097926135054346/1095929443136061502)). Were you able to build and run this locally? If you feel confident you don't need CI tests, feel free to merge. Else we can wait for @klochek to help us make sense of the CI problems.

**Review by bhackett1024 — APPROVED** [review]

File: mojo/core/node_controller.cc (hunk 1 above)
Comment (sgeisenh) [reply mojo/core/node_controller.cc:665]: There's a note above (in the original Chromium source) that justifies the copy of the `NodeName` that also seems to suggest that we might be able to make the behavior of the function more consistent by extending the lifetime of the channel.

I also added some context to the linear issue so I'll remove this from the code comment.

**Pull request merged**